Rock.Messaging.RabbitMQ/RabbitEngine.cs

```diff
@@ -0,0 +1,105 @@
+                HostName = connectionStrs[0],
+                Port = int.Parse(connectionStrs[1]),
+                VirtualHost = _config.vHost,
+            };
+            _connection = _connectionFactory.CreateConnection();
+            
+        }
+
+        public void Dispose()
+        {
+            _connection.Dispose();
+            _consumerModel.Abort();
+        }
+
+        #region Receiver
+        public void Start(string selector) // TODO: Discuss: RabbitMQ supports getting individual messages- should Rock.Messaging support that?
+        {
+            _consumerModel = _connection.CreateModel();
+            _consumerModel.ExchangeDeclare(_config.Exchange, _config.ExchangeType ?? (topical ? "topic": "direct"), true); // Should perhaps configure this default somehow? Also RabbitMQ Topics might not fulfill the same intent as the abstraction?
+            _consumerModel.QueueDeclare(_config.QueueName, true, false, false, null); // TODO: Discuss: Add config settings for these? It's already pretty cluttered...
+            if (!string.IsNullOrWhiteSpace(selector))
+            {
+                _consumerModel.QueueBind(_config.QueueName, _config.Exchange, selector);
+            }
+            else
+            {
+                _consumerModel.QueueBind(_config.QueueName, _config.Exchange, _config.RoutingKey ?? string.Empty);
+            }
+            
+            var consumer = new EventingBasicConsumer(_consumerModel);
+            consumer.Received += delegate(object sender, BasicDeliverEventArgs args)
+            {
+                var messageReceived = new MessageReceivedEventArgs(new RabbitReceiverMessage(args, _consumerModel));
+                OnMessageReceived(messageReceived);
+                if (_config.AutoAcknowledge)
+                {
+                    _consumerModel.BasicAck(args.DeliveryTag, false);
+                }
+            };
+            _consumerModel.BasicQos(0, _config.MaxRequests, true);
+            _consumerModel.BasicConsume(_config.QueueName, false, consumer);
+        }
+
+        protected virtual void OnMessageReceived(MessageReceivedEventArgs args)
+        {
+            var handler = MessageReceived;
+
+            if (handler != null)
+            {
+                handler(this, args);
+            }
+        }
+        public event EventHandler<MessageReceivedEventArgs> MessageReceived;
+
+        #endregion
+        public Task SendAsync(ISenderMessage message)
+        {
+            return Task.Run(delegate
+            {
+                using (var model = _connection.CreateModel())
+                {
+                    var props = model.CreateBasicProperties();
+                    props.Headers = message.Headers.ToDictionary<KeyValuePair<string, string>, string, object>(kvp => kvp.Key, kvp => kvp.Value); // IEnum<KVP<S,S>> in, Dictionary<string, object> out.
+                    props.ContentType = message.MessageFormat.ToString();
```

Comment:
Instead of calling `.ToString()` on the `MessageFormat` enum, perhaps a switch would be more appropriate. Each `MessageFormat` value would map to an valid MIME content type.